core/domain/collection_services.py

```diff
@@ -1098,11 +1091,8 @@ def publish_collection_and_update_user_profiles(
     rights_manager.publish_collection(committer, collection_id)  # type: ignore[no-untyped-call]
     contribution_time_msec = utils.get_current_time_in_millisecs()
     collection_summary = get_collection_summary_by_id(collection_id)
-    if collection_summary is None:
-        raise Exception(
-            'No collection summary model exists for the give id: %s'
-            % collection_id
-        )
+    # Ruling out the possibility of None for mypy type checking.
```

Comment:
As usual I think this needs an explanation in the comment of why it's impossible for collection_summary to be None here.

Reply:
I reverted the changes because I didn't find any specific validation which stops collection_summary from being None. However, if we see the docstring of this function it looks to me that collection_summary is never going to be none, but for security, I added a raise exception here.

A small conversation regarding this PR can be found here https://github.com/oppia/oppia/pull/15762#discussion_r928160110